paymentsheet/src/main/java/com/stripe/android/link/LinkActivityViewModel.kt

```diff
@@ -230,11 +230,19 @@ internal class LinkActivityViewModel @Inject constructor(
      */
     private suspend fun confirmLinkPayment(selectedPayment: LinkPaymentMethod) = runCatching {
         require(selectedPayment.readyForConfirmation()) { "LinkPaymentMethod must be ready for confirmation" }
-        linkConfirmationHandler.confirm(
-            paymentDetails = selectedPayment.details,
-            cvc = selectedPayment.collectedCvc,
-            linkAccount = requireNotNull(linkAccount) { "LinkAccount must not be null for confirmation" },
-        )
+        val linkAccount = requireNotNull(linkAccount) { "LinkAccount must not be null for confirmation" }
+        when (selectedPayment) {
+            is LinkPaymentMethod.Consumer -> linkConfirmationHandler.confirm(
+                paymentDetails = selectedPayment.details,
+                cvc = selectedPayment.collectedCvc,
+                linkAccount = linkAccount,
+            )
+            is LinkPaymentMethod.Link -> linkConfirmationHandler.confirm(
+                paymentDetails = selectedPayment.linkPaymentDetails,
+                cvc = selectedPayment.collectedCvc,
+                linkAccount = linkAccount,
+            )
```

Comment:
This naming is confusing to me. Can we make more descriptive what these two cases mean?